### PR TITLE
feat(continuous_screening): admin can dismiss all matches

### PR DESCRIPTION
**Pull request by OrriMandarin**
This pull request introduces a new "dismiss" action for continuous screenings, allowing admin users to dismiss all pending matches for a screening that is in review and in a case. It adds the necessary API endpoint, permission, repository, and usecase logic, along with accompanying tests and mock updates. The changes also refactor batch status updates for matches to improve efficiency and consistency.

**API and Usecase Enhancements:**

* Added a new API endpoint `PATCH /continuous-screenings/:id/dismiss` that allows authorized users to dismiss all pending matches for a continuous screening in review and in a case. [[1]](diffhunk://#diff-478bc9930523d2cb1072aa084f696518349c18e0db006524d4f3dfa1a413cf32R302-R326) [[2]](diffhunk://#diff-98ba00aec3a81a2ed9b83cab1c3de9ae182c63060b8d8a4a06bb443f90648e52R212-R214)
* Implemented the `DismissContinuousScreening` method in `ContinuousScreeningUsecase`, which validates permissions, updates match statuses to "skipped", and sets the screening status to "no_hit".

**Permissions and Security:**

* Introduced a new permission `CONTINUOUS_SCREENING_HIT_DISMISS` and restricted its usage to admin roles for dismissing hits on continuous screenings. [[1]](diffhunk://#diff-8ad35ab77738018a1bc9ad9bfb9aa424898e3ddc56f850b464a76193122d1185R66) [[2]](diffhunk://#diff-8ad35ab77738018a1bc9ad9bfb9aa424898e3ddc56f850b464a76193122d1185R130) [[3]](diffhunk://#diff-d3c380581ae562707f98543b91c15469243216c4bca5130f240ac39d484b8cfbR46)
* Added enforcement logic to check for the new permission when dismissing continuous screening hits. [[1]](diffhunk://#diff-f6a76f590d637d251ef285be0c3b5294b766d91c44672b3f67d6aadbc944619eR17) [[2]](diffhunk://#diff-f6a76f590d637d251ef285be0c3b5294b766d91c44672b3f67d6aadbc944619eR66-R72)

**Repository and Batch Updates:**

* Added `UpdateContinuousScreeningMatchStatusByBatch` to repositories and mocks, allowing efficient batch updates of match statuses instead of individual updates. Updated usecase and tests to use this method. [[1]](diffhunk://#diff-39633c2fb9f730ec8165dc6efe467c7ec8a39a5f5c5fd5c096237104c4f99847R466-R490) [[2]](diffhunk://#diff-b7cb723d8520b96da2715366247f7013b202a8a3384d16178e87520015dd04baR164-R174) [[3]](diffhunk://#diff-4477cb6ab44ee61b7feb332da87d6f6e8a660ef81159ca3c50ede03d36c8a2a8R96-R102)

**Testing Improvements:**

* Added comprehensive tests for the new dismiss functionality, including permission checks, status updates, and edge cases. Refactored existing tests to use batch status updates where appropriate. [[1]](diffhunk://#diff-2aabeacf638a68dbd0bff1f9598f9c96904d65b4ffe1ec44826a18f597dd31f8L193-R196) [[2]](diffhunk://#diff-2aabeacf638a68dbd0bff1f9598f9c96904d65b4ffe1ec44826a18f597dd31f8R262-R264) [[3]](diffhunk://#diff-2aabeacf638a68dbd0bff1f9598f9c96904d65b4ffe1ec44826a18f597dd31f8R495-R628)

(References: [[1]](diffhunk://#diff-478bc9930523d2cb1072aa084f696518349c18e0db006524d4f3dfa1a413cf32R302-R326) [[2]](diffhunk://#diff-98ba00aec3a81a2ed9b83cab1c3de9ae182c63060b8d8a4a06bb443f90648e52R212-R214) [[3]](diffhunk://#diff-21539c8d3fafaa3774cb1dfe6a6c478d859b6f2f8dafe0c7136652eca1981334R258-R348) [[4]](diffhunk://#diff-8ad35ab77738018a1bc9ad9bfb9aa424898e3ddc56f850b464a76193122d1185R66) [[5]](diffhunk://#diff-39633c2fb9f730ec8165dc6efe467c7ec8a39a5f5c5fd5c096237104c4f99847R466-R490)

### CODE REVIEW

File: usecases/continuous_screening/screening.go [hunk 1]
```diff
@@ -252,3 +255,94 @@ func (uc *ContinuousScreeningUsecase) createWhitelist(
...
 }
+
+// Dismiss function can only be called if the continuous screening is in review and in case and by an admin user
+// - Get the continuous screening with matches by id
+// - Check if the user has permission to dismiss the continuous screening
+// - Check if the continuous screening is in review and in case
+// - Update the match statuses to skipped
+// - Update the continuous screening status to no_hit
+// Return the continuous screening with matches
+func (uc *ContinuousScreeningUsecase) DismissContinuousScreening(ctx context.Context,
```
Comment (OrriMandarin) [inline usecases/continuous_screening/screening.go]: This new function didn't interact with Case, don't know if it needs to perform some actions on the case
